### PR TITLE
ci: add credo to ci

### PR DESCRIPTION
**Pull request by sborrazas**
Since this is the first time adding credo, when diffing it with master it will fail. But from now on it will only print any credo errors ADDED to the codebase, it won't complain about existing unchanged lines.

### CODE REVIEW

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -39,6 +39,11 @@ jobs:
       - name: Setup dependencies
         run: make docker-deps
 
+      - name: Reset master
```
Comment (thepiwo) [inline .github/workflows/test-action.yml:42]: why is that needed?

Comment (sborrazas) [reply .github/workflows/test-action.yml:42]: Right now credo only warns for the new errors added, not for the existing ones. To be able to do that it compares the errors with the errors in master. Github does not fetch master by default, so this is why we have to create it before credo runs

**Review by thepiwo — APPROVED** [review]

**Review by jyeshe — APPROVED** [review]
